lua/nvim-treesitter/configs.lua

```diff
@@ -421,6 +421,9 @@ function M.setup(user_data)
   end
 
   local ensure_installed = user_data.ensure_installed or {}
```

Comment:
I think the fix should be
```lua
local ensure_installed = user_data.ensure_installed
ensure_installed = ensure_installed and type(ensure_installed) == "table" and ensure_installed or { ensure_installed }
```

Reply:
I am unsure whether this plugin could handle "all" like a normal language, because `ensure_installed = { "all", "c", "lua", }` is meaningless. but I think it can handle `ensure_installed = "all"` at least.

Reply:
`"all"` (which is really not recommended for users, and will be heavily discouraged in 1.0) already includes the necessary parsers and should be left alone. Similarly, I don't want to support, e.g., `ensure_installed = "lua"`. So the logic should be "if (and only if) `ensure_installed` is a table, make sure it contains the mandatory parsers".

If I missed a case in the control flow, (only) this should be fixed.

Reply:
But is it a _reasonable_ control flow?